libs/checkout/src/drivers/interfaces/payment-service.interface.ts

```diff
@@ -0,0 +1,20 @@
+import { Observable } from 'rxjs';
+import { DaffCharge } from '../../models/payment/request/charge';
+import { DaffTransactionResponse } from '../../models/payment/response/transaction-response';
+
+export interface DaffPaymentServiceInterface {
```

Comment:
I think the interface here is wrong, our only consideration here is generating the payment nonce. So, a payment interface I would expect to look something like:

```ts
export interface DaffPaymentServiceInterface {
   generateToken(card: DaffCreditCard): Observable<DaffCreditCardToken>;
}
```

Authorizing and charging cards is the responsibility of a backend system.

Reply:
I redid all of this. See what you think